### PR TITLE
added valid number range

### PR DESCRIPTION
**Pull request by nikhil-kumar94**
**Title:** 
- Valid Number range plugin

**Related Issues:**
- #674

### CODE REVIEW

File: plugins/default/validNumberRange.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  PluginContext,
+  PluginHandler,
+  PluginParameters,
+} from '../types';
+import { getText } from '../utils';
+
+function validNumberRange(
+  target: Number,
+  lowerLimit: Number,
+  upperLimit: Number
```
Comment (b4s36t4) [inline plugins/default/validNumberRange.ts:12]: `Number` isn't a type. use `number` instead.

File: plugins/default/validNumberRange.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+): boolean {
+  try {
+    return target >= lowerLimit && target <= upperLimit;
+  } catch (error) {
+    return false;
+  }
+}
```
Comment (b4s36t4) [inline plugins/default/validNumberRange.ts:19]: I don't think this logic is what the expectation is. Ideally plugins are used as part of generation so we must be able to use this rule with generated content. This rule must check weather the generated content (number) is within the range.

use `getText` function to retrieve the message content and make sure the content is number (or possibly use regex to extract number from the message content) and do validation with the extracted number.

Check existing plugins here: https://github.com/Portkey-AI/gateway/tree/main/plugins/default

**Review by b4s36t4 — CHANGES_REQUESTED** [review]

**Comment by VisargD** [comment]
Hey @nikhil-kumar94 and @b4s36t4 , is there any blocker for this PR? Please let me know if you have any doubts.

**Comment by VisargD** [comment]
Closing this PR due to inactivity. Please feel free to reopen it whenever required

**Pull request closed without merging**